pandas/core/resample.py

```diff
@@ -1348,18 +1353,24 @@ def __init__(
         self.convention = convention or "E"
         self.convention = self.convention.lower()
 
-        if isinstance(loffset, str):
-            loffset = to_offset(loffset)
-        self.loffset = loffset
-
         self.how = how
         self.fill_method = fill_method
         self.limit = limit
-        self.base = base
+
+        self.origin = Timestamp(origin) if origin is not None else None
+        self.offset = Timedelta(offset) if offset is not None else None
+        if base and isinstance(freq, Tick):
+            # this conversion handle the default behavior of base
+            # and the special case of GH #10530
```

Comment:
Yes, to handle `base` argument when resampling TimedeltaIndex, `base` is treated as a pure offset. So with a frequency of `5s`, a base of `7` will be treated as an offset of 2s in the normal case (since we are doing a modulo on the frequency of `5s`), but it will treated as a pure offset (so `7s`) for TimedeltaIndex as you can see here: https://github.com/pandas-dev/pandas/commit/433602022bfbb9ee2424b77694c60f276065dee8#diff-e21e44fc4ab14efc2f09e0dacdd79fc6R241

Reply:
I just added a comment in the code to explain that